frontend/catalyst/passes.py

```diff
@@ -304,6 +318,49 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
+def apply_pass(pass_name, *flags, **valued_options):
+    """Applies a single pass to the qnode"""
+
+    def decorator(qnode):
+
+        if not isinstance(qnode, qml.QNode):
+            # Technically, this apply pass is general enough that it can apply to
+            # classical functions too. However, since we lack the current infrastructure
+            # to denote a function, let's limit it to qnodes
+            raise TypeError(f"A QNode is expected, got the classical function {qnode}")
+
+        def qnode_call(*args, **kwargs):
+            pass_pipeline = kwargs.get("pass_pipeline", [])
+            pass_pipeline.append(Pass(pass_name, *flags, **valued_options))
+            kwargs["pass_pipeline"] = pass_pipeline
+            return qnode(*args, **kwargs)
+
+        return qnode_call
+
+    return decorator
+
+
+def apply_pass_plugin(plugin_name, pass_name, *flags, **valued_options):
```

Comment:
Is it possible to reuse `apply_pass` here in the `apply_pass_plugin`? I think the only difference is the thing being appened to the `pass_pipeline` right?

Reply:
The difference is the arity of the function. We could have a special kwarg / default name for the plugin path, but that would change the meaning of `**valued_options`. In other words, there could not be a valued option with the name we choose for the kwarg. :thinking: